test/tests/pbs_smoketest.py

```diff
@@ -985,7 +985,9 @@ def test_job_sort_formula_threshold(self):
         msg = "Checking the job state of %s, runs after %s is deleted" % (j3id,
                                                                           j4id)
         self.logger.info(msg)
-        self.server.deljob(id=j4id, wait=True)
+        attribs = self.server.status(JOB, 'job_state', id=j4id)
+        if attribs[0]['job_state'] == 'R':
+            self.server.deljob(id=j4id, wait=True)
```

Comment:
I think the fix should be in self.server.deljob() to handle a case when the job actually exited/no longer exist, and should not hang...

Reply:
@bayucan It does not hang. But it throws unknown job id error as qdel would if the job didnt exist.

Reply:
@visheshh : Then how about just doing a try...else around self.server.deljob() and just allow for the "Unkonwn jobid" condition? Even if you check if the job's state is R, what if by the time the self.server.deljob() executes, job could have disappeared quickly...

Reply:
@bayucan Thanks for the suggestion. I will do that.

Reply:
I'm not sure we want to have unknown job id not throw an exception.  I mean it IS an error.  It just happens that some of our tests don't care.  I think it should be up to the test to catch the exception if they don't care.  Either that or pass an argument that you don't care about it.  In any case, using a walltime of 4s is too small for our slow testbed machines.

Reply:
This also won't help if the job is in R state when you status it, but is deleted by the time you deljob it. 

I agree with bhroam's suggestion of passing an argument if you want to throw an exception if the jobid doesn't exit.